src/runtime/task/mod.rs

```diff
@@ -404,9 +398,21 @@ pub(crate) struct ParkState {
 
 /// A `TaskId` is a unique identifier for a task. `TaskId`s are never reused within a single
 /// execution.
-#[derive(PartialEq, Eq, Hash, Clone, Copy, PartialOrd, Ord, Debug)]
+#[derive(PartialEq, Eq, Hash, Clone, Copy, PartialOrd, Ord)]
 pub struct TaskId(pub(super) usize);
 
+impl Debug for TaskId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        TASK_ID_TO_TAGS.with(|cell| {
+            let map = cell.borrow();
+            match map.get(self) {
+                Some(tag) => f.write_str(&format!("{:?}", tag)),
+                None => f.debug_tuple("TaskId").field(&self.0).finish(),
```

Comment:
We always want to print the task ID, even if a tag is set.

Reply:
Hmm, yes and no. In a "properly tagged" run the TaskId is noise. I can turn it back on, dunno if there is a case to be made to have this toggleable. This was toggleable in the previous scheme.

Reply:
I agree with James -- I think we should always print the task id.  Don't even need to make it optional; just always print it.  Otherwise, if I forget to set a tag, it'll be hard to figure out.